diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ce7c56d..aa97c485 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,7 @@ if(NOT COMMAND add_llvm_library)
         # add -std=gnu++14
         set(CMAKE_CXX_EXTENSIONS ON)

-        set(COMMON_FLAGS "-fPIC -Werror -Wall")
+        set(COMMON_FLAGS "-fPIC")

         if(CMAKE_BUILD_TYPE MATCHES "Debug")
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COMMON_FLAGS} -O0 -fno-rtti")